deps/key-value-storage/src/lib.rs

```diff
@@ -27,10 +27,15 @@ pub struct SetParameters {
     pub overwrite: bool,
 }
 
+pub enum SetResult {
+    Inserted,
+    AlreadyExists,
+}
```

Comment:
Some std stuff uses an option for this i.e. return Some(val) if val was already there. Might be simpler. Up to you.